tests/test_wrappers.py

```diff
@@ -417,9 +417,12 @@ class ClassWithCallback(wrappers.KerasClassifier):
     """Must be defined at top level to be picklable.
     """
 
-    def __init__(self, **sk_params):
-        self.callbacks = [SentinalCallback()]
-        super().__init__(**sk_params)
+    def __init__(self, callbacks=None, hidden_dim=(100,)):
+        if callbacks is None:
+            self.callbacks = [SentinalCallback()]
+        else:
+            self.callbacks = callbacks
```

Comment:
I think this brings up an interesting question that is not a blocker for this PR but will be interesting for param routing: a quite common pattern in scikit-learn estimators is to parse/modify an `__init__` param in `fit`, ex:
```python
def fit(X, y, ...):
    self.random_state_ = np.random.RandomState(seed=self.random_state)
```

Where downstream the estimator will only use the `self.random_state_` param and not `self.random_state`.

I think it would be nice to allow users to do something like:

```python3
class MyClassifier(KerasClassifier):

    def hook_params(self):   # in base class this would just be a pass
         if self.batch_size is None:
              self.batch_size_ = 32
         else:
              self.batch_size_ = self.batch_size
```

But the trick will then be routing that parameter so that `callbacks_` always overrides `callbacks`. Maybe if two parameters match except for `_`'s at the end, the one with the most `_` takes precedence?  So that:

```python3
def build_fn(batch_size=32):
    print(f"batch_size={batch_size}")
    ...

class MyClassifier(KerasClassifier):

    def hook_params(self):   # in base class this would just be a pass
         if self.batch_size is None:
              self.batch_size_ = 32
         else:
              self.batch_size_ = self.batch_size
         self.batch_size__ = 64

estimator = MyClassifier()
estimator.fit(X, y)  # always prints `batch_size=64`
```

Reply:
Fyi, sklearn.utils has a "check_random_state" function. I don't recall estimators ever setting a "random_state_" attribute.

Reply:
Yeah, I was just looking for a clear example, I didn't check that it's a real one.

Reply:
I updated it to a real-world example from Scikit-Learn.

Reply:
Thanks for the update; I also do that in my estimators.

Reply:
Again, not for this PR, but certainly a required feature for parameter routing. I'm going to copy this discussion to #50 .